scripts/auth0connections/auth0connections.py

```diff
@@ -95,13 +120,31 @@ def create(ctx):
         if not connection_name in remote_connections:
             click.echo(f"Creating {connection_name}")
             resp = client.connections.create(body)
-            click.echo(pprint(resp))
+            if resp:
+                click.echo(pprint(resp))
         else:
             click.echo(
                 f"Skipping: {connection_name} as it already exists. Delete it "
                 f"from auth0 if you want this script to recreate it"
             )
 
 
+@cli.command()
+@click.pass_context
+@click.argument('name')
+def delete(ctx, name):
+    click.echo(f"Deleting connection {name}")
+    client = get_client()
+    remote_connections = dict((c["name"], c) for c in client.connections.all())
+    try:
+        connection_id = remote_connections[name]['id']
+    except KeyError:
+        click.echo(f"Error: Connection {name} does not exist (remotely)", err=True)
+        sys.exit(1)
```

Comment:
The `try`/`except` is probably overkill here, the name not being there wouldn't be that exceptional.
Probably something like:

```python
if name not in remote_connections:
    click.echo(f"Error: ...")
    sys.exit()
```

seems less scary ☺️ 

Also, I've seen this `dict((key, value) for ...` earlier and I was puzzled, maybe it's just my bad python but it took me a bit to realise it was really building a dictionary from the loop. Any reason to not use something like:

```python
remote_connections = { c["name"]: c for c in client.connections.all() }
```

I don't mind really, it's just that the sight of curly braces is a strong cue for the fact we're building a dictionary which may help slow people like me ☺️ 

Lastly, as far as I can tell we're building a dictionary her to get the `id` of the connection to delete right? If that's the case maybe you could avoid building this dictionary and just search for the client with the given name. 
But to be fair I'm not sure that may simplify the code much as you would still have to have a for with an `if` and a `break` - so maybe build a dictionary is less ugly. 
Unless we have a huge amount of connections it wouldn't make much difference memory-wise so probably building the dictionary is good enough.

Reply:
So it would be:
```python
    try:
        connection_id = remote_connections[name]['id']
    except KeyError:
        click.echo(f"Error: Connection {name} does not exist (remotely)", err=True)
        sys.exit(1)
```
vs
```python
if name not in remote_connections:
    click.echo(f"Error: Connection {name} does not exist (remotely)", err=True)
    sys.exit(1)
connection_id = remote_connections[name]['id']
```
I think either is fine. I'm not sure there is any python style guidance on only using try/except when the unhappy path is really 'exceptional', is there? I like try/except because it allows the reader to ignore the unhappy path when skimming.

+1 for using click.echo

I got the impression people preferred `dict((key, value) for ...` over `{key: value for ...}` but will be interested if you can find a reference for this.

I guess these are style things. Maybe @ntoll can suggest from his experience?